lib/matplotlib/colors.py

```diff
@@ -286,6 +286,11 @@ def to_rgba(c, alpha=None):
         Tuple of floats ``(r, g, b, a)``, where each channel (red, green, blue,
         alpha) can assume values between 0 and 1.
     """
+    if isinstance(c, tuple) and len(c) == 2 and isinstance(c[0], str):
+        if 0 <= c[1] <= 1:
+            c, alpha = c[0], c[1]
+        else:
+            raise ValueError('Alpha must be between 0 and 1, inclusive.')
```

Comment:
Afaict this check is redundant with the one in _to_rgba_no_colorcycle and should be removed.